### PR TITLE
Saner defaults for configs

### PR DESCRIPTION
**Pull request by gouthamve**
Signed-off-by: Goutham Veeramachaneni <gouthamve@gmail.com>

**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by bboreham** [comment]
Seems reasonable, but I don't understand why you'd want so many tokens.
Also, what happens if you raise the number without noticing?  Say you had 4 old ingesters with 128 tokens and you add a new one with 512 - does it get half the series?

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,16 @@
...
 * [CHANGE] Lifecycler component now enters Failed state on errors, and doesn't exit the process. (Important if you're vendoring Cortex and use Lifecycler) #2251
 * [CHANGE] `/ready` handler now returns 200 instead of 204. #2330
+* [CHANGE] Better defaults for the following options: 
+  - `-<prefix>.consul.consistent-reads`: Old default: `true`, new default: `false`. This reduces the load on Consul.
+  - `-<prefix>.consul.watch-rate-limit`: Old default: 0, new default: 1. This rate limits the reads to 1 per second. Which is good enough for ring watches.
+  - `-distributor.health-check-ingesters`: Old default: `false`, new default: `true`.
+  - `-ingester.max-stale-chunk-idle`: Old default: 0, new default: 2m. This lets us expire series that we know are stale early.
+  - `-ingester.num-tokens`: Old default: 128, new default: 128. This allows for better load distribution in large clusters.
+  - `-ingester.spread-flushes`: Old default: false, new default: true. This allows to better de-duplicate data and use less space.
```
Comment (bboreham) [inline CHANGELOG.md]: This is not actually changed in the code.  Also you'd want to set `ChunkAgeJitter` to 0.

**Comment by bboreham** [comment]
Should update the docs to remove advice to set these, or at least note it's only for older versions. 
 
Would we take these out of the example manifests?

**Comment by tomwilkie** [comment]
> Seems reasonable, but I don't understand why you'd want so many tokens.

More tokens == better load balancing - less variation is the size of the ranges owned by the ingesters.

> Also, what happens if you raise the number without noticing? Say you had 4 old ingesters with 128 tokens and you add a new one with 512 - does it get half the series?

More tokens will mean more ranges, but those ranges will be smaller, so its not quite linear IIRC.

**Comment by bboreham** [comment]
That’s a fine theory, but do we have any evidence?  Has anyone ever noticed an imbalance (except me, running with 512 tokens per ingester)?

I think giving one new ingester a large fraction of the series is likely to be disastrous. Suggest this particular change should be left in the documentation.

File: CHANGELOG.md [hunk 2]
```diff
@@ -42,6 +42,16 @@
 * [CHANGE] Frontend worker in querier now starts after all Querier module dependencies are started. This fixes issue where frontend worker started to send queries to querier before it was ready to serve them (mostly visible when using experimental blocks storage). #2246
 * [CHANGE] Lifecycler component now enters Failed state on errors, and doesn't exit the process. (Important if you're vendoring Cortex and use Lifecycler) #2251
 * [CHANGE] `/ready` handler now returns 200 instead of 204. #2330
+* [CHANGE] Better defaults for the following options: 
+  - `-<prefix>.consul.consistent-reads`: Old default: `true`, new default: `false`. This reduces the load on Consul.
+  - `-<prefix>.consul.watch-rate-limit`: Old default: 0, new default: 1. This rate limits the reads to 1 per second. Which is good enough for ring watches.
+  - `-distributor.health-check-ingesters`: Old default: `false`, new default: `true`.
+  - `-ingester.max-stale-chunk-idle`: Old default: 0, new default: 2m. This lets us expire series that we know are stale early.
+  - `-ingester.num-tokens`: Old default: 128, new default: 128. This allows for better load distribution in large clusters.
```
Comment (weeco) [inline CHANGELOG.md]: Ne default is 512 not 128

File: CHANGELOG.md [hunk 3]
```diff
@@ -42,6 +42,16 @@
 * [CHANGE] Frontend worker in querier now starts after all Querier module dependencies are started. This fixes issue where frontend worker started to send queries to querier before it was ready to serve them (mostly visible when using experimental blocks storage). #2246
 * [CHANGE] Lifecycler component now enters Failed state on errors, and doesn't exit the process. (Important if you're vendoring Cortex and use Lifecycler) #2251
 * [CHANGE] `/ready` handler now returns 200 instead of 204. #2330
+* [CHANGE] Better defaults for the following options: 
+  - `-<prefix>.consul.consistent-reads`: Old default: `true`, new default: `false`. This reduces the load on Consul.
+  - `-<prefix>.consul.watch-rate-limit`: Old default: 0, new default: 1. This rate limits the reads to 1 per second. Which is good enough for ring watches.
```
Comment (pracucci) [inline CHANGELOG.md]: Where is this change in the code? I'm missing it.

File: CHANGELOG.md (hunk 2 above)
Comment (pracucci) [inline CHANGELOG.md]: New default is `512`

File: CHANGELOG.md [hunk 4]
```diff
@@ -42,6 +42,16 @@
...
+  - `-distributor.health-check-ingesters`: Old default: `false`, new default: `true`.
+  - `-ingester.max-stale-chunk-idle`: Old default: 0, new default: 2m. This lets us expire series that we know are stale early.
+  - `-ingester.num-tokens`: Old default: 128, new default: 128. This allows for better load distribution in large clusters.
+  - `-ingester.spread-flushes`: Old default: false, new default: true. This allows to better de-duplicate data and use less space.
+  - `-<prefix>.memcached.batchsize`: Old default: 0, new default: 1024. This allows batching of requests and keeps the concurrent requests low.
```
Comment (pracucci) [inline CHANGELOG.md]: Where is this change in the code?

File: CHANGELOG.md [hunk 5]
```diff
@@ -42,6 +42,16 @@
 * [CHANGE] Frontend worker in querier now starts after all Querier module dependencies are started. This fixes issue where frontend worker started to send queries to querier before it was ready to serve them (mostly visible when using experimental blocks storage). #2246
 * [CHANGE] Lifecycler component now enters Failed state on errors, and doesn't exit the process. (Important if you're vendoring Cortex and use Lifecycler) #2251
 * [CHANGE] `/ready` handler now returns 200 instead of 204. #2330
+* [CHANGE] Better defaults for the following options:
```
Comment (pracucci) [inline CHANGELOG.md]: Could you reference the PR number too, please? _It's used to simplify finding PRs missing in the changelog during the release process._

**Comment by bboreham** [comment]
Do we want to remove recommendations from the docs for what is now the default?
E.g. https://github.com/cortexproject/cortex/blame/b3112b3315da58eb0ce3b50caa8df05f0a633db8/docs/guides/running.md#L244-L248

File: docs/configuration/arguments.md [hunk 6]
```diff
@@ -282,10 +282,6 @@ It also talks to a KVStore and has it's own copies of the same flags used by the
 
   If a series receives a [staleness marker](https://www.robustperception.io/staleness-and-promql), then we wait for this duration to get another sample before we close and flush this series, removing it from memory. You want it to be at least 2x the scrape interval as you don't want a single failed scrape to cause a chunk flush.
 
-- `-ingester.chunk-age-jitter`
```
Comment (bboreham) [inline docs/configuration/arguments.md]: it wasn't really necessary to delete this

**Pull request merged**